src/brains2/src/estimation/spline_test.cpp

```diff
@@ -0,0 +1,456 @@
+    n_blue = 0;
+    n_yellow = 0;
+
+    for (size_t i = 0; i < cones_type.size(); ++i) {
+        if (cones_type[i] == "blue") {
+            blue_cones(n_blue, 0) = cones_x[i];
+            blue_cones(n_blue, 1) = cones_y[i];
+            ++n_blue;
+
+        } else if (cones_type[i] == "yellow") {
+            yellow_cones(n_yellow, 0) = cones_x[i];
+            yellow_cones(n_yellow, 1) = cones_y[i];
+            ++n_yellow;
+        }
+    }
+
+    // print the cones
+    // std::cout << "Blue cones (X, Y):\n";
+    // for (size_t i = 0; i < blue_cones.rows(); ++i) {
+    //     std::cout << "(" << blue_cones(i, 0) << ", " << blue_cones(i, 1) << ")\n";
+    // }
+
+    // std::cout << "\nYellow cones (X, Y):\n";
+    // for (size_t i = 0; i < yellow_cones.rows(); ++i) {
+    //     std::cout << "(" << yellow_cones(i, 0) << ", " << yellow_cones(i, 1) << ")\n";
+    // }
+
+    // For demonstration, we'll create a path by averaging the positions of blue and yellow cones.
+    // This assumes that cones are placed in pairs along the track.
+
+    auto start = std::chrono::high_resolution_clock::now();
+
+    double initial_heading_yellow = atan2(yellow_cones(1, 1) - yellow_cones(0, 1),
+                                          yellow_cones(1, 0) - yellow_cones(0, 0));
+    double initial_heading_blue = atan2(blue_cones(1 , 1) - blue_cones(0, 1),
+                                        blue_cones(1, 0) - blue_cones(0, 0));
+    double final_heading_yellow = atan2(yellow_cones(yellow_cones.rows() - 1, 1) - yellow_cones(yellow_cones.rows() - 2, 1),
+                                        yellow_cones(yellow_cones.rows() - 1, 0) - yellow_cones(yellow_cones.rows() - 2, 0));
+    double final_heading_blue = atan2(blue_cones(blue_cones.rows() - 1, 1) - blue_cones(blue_cones.rows() - 2, 1),
+                                      blue_cones(blue_cones.rows() - 1, 0) - blue_cones(blue_cones.rows() - 2, 0));
+
+    MatrixPair splines_blue = fit_open_spline(blue_cones, curv_weight, initial_heading_blue, final_heading_blue);
+    MatrixPair splines_yellow = fit_open_spline(yellow_cones, curv_weight, initial_heading_yellow, final_heading_yellow);
+
+    Eigen::VectorXd delta_s_blue =
+        compute_spline_interval_lengths(splines_blue.first, splines_blue.second);
+    Eigen::VectorXd delta_s_yellow =
+        compute_spline_interval_lengths(splines_yellow.first, splines_yellow.second);
+
+    auto result_blue = uniformly_sample_spline(splines_blue.first, splines_blue.second, delta_s_blue, resample_points);
+    
+    auto result_yellow = uniformly_sample_spline(splines_yellow.first, splines_yellow.second, delta_s_yellow, resample_points);
+
+    Eigen::VectorXd X_interp_blue = std::get<0>(result_blue);
+    Eigen::VectorXd Y_interp_blue = std::get<1>(result_blue);
+    Eigen::VectorXi idx_interp_blue = std::get<2>(result_blue);
+    Eigen::VectorXd t_interp_blue = std::get<3>(result_blue);
+    Eigen::VectorXd s_interp_blue = std::get<4>(result_blue);
```

Comment:
You can use structured bindings here https://en.cppreference.com/w/cpp/language/structured_binding